### PR TITLE
hm 13. Spring boot and data

### PR DESCRIPTION
**Pull request by borysovdenys**

### CODE REVIEW

File: Borysov/quiz-boot/src/main/java/com/quiz/borysov/quizboot/config/QuizConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import org.springframework.orm.hibernate5.LocalSessionFactoryBean;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.sql.DataSource;
+
+@Configuration
+public class QuizConfiguration {
+}
```
Comment (dinamio) [inline Borysov/quiz-boot/src/main/java/com/quiz/borysov/quizboot/config/QuizConfiguration.java:20]: Why you need this empty configuration class?

File: Borysov/quiz-boot/src/main/java/com/quiz/borysov/quizboot/controller/UserController.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        if (loginError != null) {
+            model.addAttribute("errorLogIn", "errorLogIn");
+            session.removeAttribute("message");
+        }
+        return "index";
+    }
+
+    @RequestMapping(method = POST, value = "login")
+    public String doLogin(WebRequest req, HttpSession session){
+        LOGGER.info("Log in");
```
Comment (dinamio) [inline Borysov/quiz-boot/src/main/java/com/quiz/borysov/quizboot/controller/UserController.java:59]: Why you need this? As far as I understand, you doing login through spring security

File: Borysov/quiz-boot/src/main/java/com/quiz/borysov/quizboot/controller/UserController.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+            LOGGER.error("Login or password entered incorrectly!", e);
+            session.setAttribute("errorLogIn", "errorLogIn");
+            return "redirect:login-get";
+        }
+    }
+
+    @RequestMapping(method = GET, value = "logout")
+    public String doLogout(HttpSession session){
+        LOGGER.info("Log out");
+        session.invalidate();
```
Comment (dinamio) [inline Borysov/quiz-boot/src/main/java/com/quiz/borysov/quizboot/controller/UserController.java:79]: Same question here

**Review by dinamio — APPROVED** [review]
Looks good

File: Borysov/quiz-boot/src/main/java/com/quiz/borysov/quizboot/controller/UserController.java (hunk 2 above)
Comment (borysovdenys) [reply Borysov/quiz-boot/src/main/java/com/quiz/borysov/quizboot/controller/UserController.java:59]: I get it already.

File: Borysov/quiz-boot/src/main/java/com/quiz/borysov/quizboot/controller/UserController.java (hunk 3 above)
Comment (borysovdenys) [reply Borysov/quiz-boot/src/main/java/com/quiz/borysov/quizboot/controller/UserController.java:79]: understood

File: Borysov/quiz-boot/src/main/java/com/quiz/borysov/quizboot/config/QuizConfiguration.java (hunk 1 above)
Comment (borysovdenys) [reply Borysov/quiz-boot/src/main/java/com/quiz/borysov/quizboot/config/QuizConfiguration.java:20]: I will delete it

**Pull request merged**